libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -319,6 +322,80 @@ void AP_UAVCAN::init(uint8_t driver_index, bool enable_filters)
     debug_uavcan(2, "UAVCAN: init done\n\r");
 }
 
+// send ESC telemetry messages over MAVLink
+void AP_UAVCAN::send_esc_telemetry_mavlink(uint8_t mav_chan)
+{
+
+    WITH_SEMAPHORE(_esc_sem);
+
+    bool group[3] = {false};
```

Comment:
Maybe just do this inline instead of as a separate step; form up the boolean as part of handling the data for each ESC.

Reply:
If the purpose of this is a gain in performance, I think this is very negligible and it doesn't worth the penalty in terms of code readability